### PR TITLE
[WIP] Some clang-tidy fixes

### PR DESCRIPTION
**Pull request by aarlt**
PR contains some commits that apply different clang-tidy fixes that may be useful.

### CODE REVIEW

File: libevmasm/CommonSubexpressionEliminator.h [hunk 1]
```diff
@@ -130,7 +130,7 @@ class CSECodeGenerator
 	void generateClassElement(Id _c, bool _allowSequenced = false);
 	/// @returns the position of the representative of the given id on the stack.
 	/// @note throws an exception if it is not on the stack.
-	int classElementPosition(Id _id) const;
+	[[nodiscard]] int classElementPosition(Id _id) const;
```
Comment (chriseth) [inline libevmasm/CommonSubexpressionEliminator.h:133]: What is the idea behind marking this function nodiscard?

**Review by chriseth — CHANGES_REQUESTED** [review]
This PR is too big. Please submit individual PRs for each change.

File: libevmasm/CommonSubexpressionEliminator.h (hunk 1 above)
Comment (aarlt) [reply libevmasm/CommonSubexpressionEliminator.h:133]: the idea is just that if a function returns something, it need to be consumed. i'm not sure whether it makes sense to do this everywhere - probably not.

**Comment by aarlt** [comment]
@chriseth I created individual PR's out of this PR. Let's see what PR makes sense to merge.

**Comment by aarlt** [comment]
replaced by #8568 #8569 #8570 #8571

**Pull request closed without merging**